spam/spam_tutorial.py

```diff
@@ -140,234 +140,400 @@
+# We'll have to tune the output of our LFs based on the TextBlob scores.
+# Tuning input parameters or thresholds from model outputs is a common practice in developing LFs.
+
+# %%
+import matplotlib.pyplot as plt
+
+spam_polarities = [
+    textblob_sentiment(x).polarity for _, x in df_dev.iterrows() if x.label == SPAM
+]
+
+ham_polarities = [
+    textblob_sentiment(x).polarity for _, x in df_dev.iterrows() if x.label == HAM
+]
 
-buckets = error_buckets(golds=Y_dev, preds=L_dev_array)
+plt.hist([spam_polarities, ham_polarities])
+plt.title("TextBlob sentiment polarity scores")
+plt.xlabel("Sentiment polarity score")
+plt.ylabel("Count")
+plt.legend(["Spam", "Ham"])
+plt.show()
 
-df_dev[["text", "label"]].iloc[buckets[(SPAM, HAM)]].head()
 
 # %% [markdown]
-# On the other hand, `buckets[(SPAM, SPAM)]` points to `SPAM` data points that the LF labeled correctly.
+# Using our accuracy-over-coverage principle above, we'll target the high polarity bin on the far right in our LF since there are many more `HAM` comments.
 
 # %%
-df_dev[["text", "label"]].iloc[buckets[(SPAM, SPAM)]].head()
+@labeling_function(preprocessors=[textblob_sentiment])
+def textblob_polarity(x):
+    return HAM if x.polarity > 0.9 else ABSTAIN
+
 
 # %% [markdown]
-# And `buckets[(ABSTAIN, SPAM)]` points to data points that the LF abstained on that are actually `SPAM`.
-# Many of these will be best captured by a separate LF, but browsing these examples can be a good check that your LF is capturing most of the examples that you intended it to.
+# Let's do the same for the subjectivity scores.
+# This will run faster than the last cell, since we memoized the `Preprocessor` outputs.
 
 # %%
-df_dev[["text", "label"]].iloc[buckets[(ABSTAIN, SPAM)]].head()
+spam_subjectivities = [
+    textblob_sentiment(x).subjectivity for _, x in df_dev.iterrows() if x.label == SPAM
+]
+
+ham_subjectivities = [
+    textblob_sentiment(x).subjectivity for _, x in df_dev.iterrows() if x.label == HAM
+]
+
+plt.hist([spam_subjectivities, ham_subjectivities])
+plt.title("TextBlob sentiment subjectivity scores")
+plt.xlabel("Sentiment subjectivity score")
+plt.ylabel("Count")
+plt.legend(["Spam", "Ham"])
+plt.show()
+
+plt.hist([spam_subjectivities, ham_subjectivities], bins=[0, 0.5, 1])
+plt.title("TextBlob sentiment subjectivity scores")
+plt.xlabel("Sentiment subjectivity score")
+plt.ylabel("Count")
+plt.legend(["Spam", "Ham"])
+plt.show()
 
 
 # %% [markdown]
-# Looking at all these examples, we notice that much of the time when "my" is used, it's referring to "my channel". We can update our LF to see how making this change affects accuracy and coverage.
+# It looks like subjectivity scores above 0.5 will work pretty well for identifying `HAM` comments, though not perfectly.
+# Using this threshold alone, there's no way to create a more accurate (but still general) rule for targeting `HAM`.
```

Comment:
are you sure that there isn't a single other threshold that would result in higher accuracy? maybe lighten the claim.